### PR TITLE
Fix upgrade update stages order

### PR DESCRIPTION
**Pull request by nadavkav**
Trying to upgrade an old version of the module, I got fatal errors missing "tincanlaunch_lrs" table.

Upgrade updates should be added at the end of the file after each savepoint.
Moving first 3 IF blocks updates to the end of the file and reordering them by versions

### CODE REVIEW

**Comment by garemoko** [comment]
Huh. I always added them at the top. Good to know.

File: db/upgrade.php [hunk 1]
```diff
@@ -182,6 +115,73 @@ function xmldb_tincanlaunch_upgrade($oldversion) {
         upgrade_mod_savepoint(true, 2015033100, 'tincanlaunch');
     }
 
+    if ($oldversion < 2015112702) { //New version in version.php
```
Comment (garemoko) [inline db/upgrade.php:118]: Can you explain why this date changes?

File: db/upgrade.php [hunk 2]
```diff
@@ -182,6 +115,73 @@ function xmldb_tincanlaunch_upgrade($oldversion) {
...
+        // Adding indexes to table tincanlaunch_lrs.
+        $table->add_index('tincanlaunch_credentialid', XMLDB_INDEX_NOTUNIQUE, array('credentialid'));
+        $table->add_index('tincanlaunch_tincanlaunchid', XMLDB_INDEX_NOTUNIQUE, array('tincanlaunchid'));
+
+        // Conditionally launch create table for tincanlaunch_lrs.
+        if (!$dbman->table_exists($table)) {
+            $dbman->create_table($table);
+        }
+
+        upgrade_mod_savepoint(true, 2016021508, 'tincanlaunch');
```
Comment (garemoko) [inline db/upgrade.php:182]: And this one?

File: db/upgrade.php (hunk 1 above)
Comment (nadavkav) [inline db/upgrade.php:118]: It is co-related to the module's version in which there was a major change in the related database tables of the module.
so no matter what version you are in, if you upgrade you will apply only the changes that are relevant to you and the version you are upgrading from.

Not all regular version changes of the module needs to have a corresponding IF statement with a "version date", only those that change the DB.

Comment (garemoko) [inline db/upgrade.php:118]: I get that, but why did you change it in this PR? Presumably I had it set to the wrong version?

Are you sure this was the cause of your problem and not the fact that i hadn't updated intsall.xml's version number?  https://github.com/garemoko/moodle-mod_tincanlaunch/issues/93

File: db/upgrade.php (hunk 2 above)
Comment (nadavkav) [inline db/upgrade.php:182]: Moodle remembers the date (version) that last changed anything important in the DB structure of each module, as not on each module version there is a change in DB structure. so the $oldversion actually holds that value.

So, you can regularly change the date/version on the version.php file only, and when you have a change in DB structure you also need to use upgrade_mod_savepoint() here too.

Here is a short explanation: 
https://docs.moodle.org/dev/Activity_modules#upgrade.php
And a more detailed one:
https://docs.moodle.org/dev/Upgrade_API#Introduction

File: db/upgrade.php (hunk 1 above)
Comment (nadavkav) [inline db/upgrade.php:118]: I did not. just copy pasted your code and re order it. (indeed weird)

install.xml is only used when anyone first installs the module. and as I was upgrading, Moodle did not even read it.
of course, you should keep it updated at all times, for the sake of anyone who installs the plugin for the first time.

Comment (nadavkav) [inline db/upgrade.php:118]: Anyways, the order of the versions now seems OK.don't you agree?

Comment (nadavkav) [inline db/upgrade.php:118]: Just saw a commit by Avi Levi 
https://github.com/garemoko/moodle-mod_tincanlaunch/commit/5a9590f7f24825dca657976fd93333326c2a0229
(that works on my team, and which is currently upgrading all of the modules we use to Moodle 3.0), changed a version on the update.php file. Is that might be related?

Comment (garemoko) [inline db/upgrade.php:118]: Good point. I should have just checked the final file. Merging.
